src/core/generator/from_proto.rs

```diff
@@ -33,40 +39,102 @@ impl Context {
             namespace: Default::default(),
             config: Default::default(),
             map_types: Default::default(),
+            source_code_info: None,
         }
     }
 
+    /// Sets source code information for preservation of comments.
+    fn with_source_code_info(mut self, source_code_info: SourceCodeInfo) -> Self {
+        self.source_code_info = Some(source_code_info);
+        self
+    }
+
+    /// Retrieves comments associated with the source code information.
+    fn get_comments(&self, path: &[i32]) -> Option<String> {
```

Comment:
wdyt about moving `get_comments` to `PathBuilder` and rename it to actually `CommentsBuilder`? so we'll have separated entity to manage comments